bin/extract_utils.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import gzip
+import pyfastx
+import argparse
+import json
+from datetime import datetime
+from collections import defaultdict
+from pathlib import Path
+
+
+def mean(l):
```

Comment:
This doesn't need changing really but in future perhaps consider using the standard library for things like this e.g. `statistics.mean()`. But this is a more aesthetic concern even though it would be slightly faster due to being written in cython.